skimage/filters/rank/generic.py

```diff
@@ -92,39 +92,40 @@ def _handle_input(image, selem, out, mask, out_dtype=None, pixel_size=1):
     is_8bit = image.dtype in (np.uint8, np.int8)
 
     if is_8bit:
-        max_bin = 255
+        n_bins = 256
```

Comment:
Is this correct?

Reply:
I totally should have split this PR in two steps, but they are so intertwined.

In most of the cython code, what is called `max_bin` is actually the old `255` number `+1`.